### ISSUE TITLE
Cookie parsing mis-parses when missing equals sign

### ISSUE THREAD

**Issue by joshma** [pallets/werkzeug#841]
Chrome and Firefox let you set cookies that don't have an equals sign:

```
document.cookie = "foo";
document.cookie = "session=my-flask-session";
document.cookie; // "foo; bar=my-flask-session;"
```

This breaks `_cookie_re` in https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/_internal.py#L46 - it treats `foo; bar` as the name of the cookie, since there's no `=`, causing both `foo` and `bar` to be missing from the request.

This might be the browser being non-compliant (I think https://tools.ietf.org/html/rfc6265#section-4.1.1 requires `=` and I've filed an issue at https://code.google.com/p/chromium/issues/detail?id=572541), but given the behavior it'd be nice for Flask to handle it.

**Comment by qlans** [comment]
Year, i have met the same question. Some cookie begins with `;`, the followed by zero or more name-value-pair. The first `;` as the name of the cookie, the first name is missing.

**Comment by davidism** [comment]
Fixed in d4087a3072c8387d2d4dd57f988d051a2e202b25
